src/settlement-chain/PayerReportManager.sol

```diff
@@ -0,0 +1,381 @@
+    function getPayerReports(uint32 originatorNodeId_) external view returns (PayerReport[] memory payerReports_) {
+        return _getPayerReportManagerStorage().payerReportsByOriginator[originatorNodeId_];
+    }
+
+    /// @inheritdoc IPayerReportManager
+    function getPayerReport(
+        uint32 originatorNodeId_,
+        uint256 payerReportIndex_
+    ) external view returns (PayerReport memory payerReport_) {
+        return _getPayerReportManagerStorage().payerReportsByOriginator[originatorNodeId_][payerReportIndex_];
+    }
+
+    /* ============ Internal View/Pure Functions ============ */
+
+    function _name() internal pure override returns (string memory name_) {
+        return "PayerReportManager";
+    }
+
+    function _version() internal pure override returns (string memory version_) {
+        return "1";
+    }
+
+    /**
+     * @dev    Returns the EIP-712 digest for a payer report.
+     * @param  originatorNodeId_ The id of the node originator.
+     * @param  attesterNodeId_   The id of the node attester.
+     * @param  startSequenceId_  The start sequence ID.
+     * @param  endSequenceId_    The end sequence ID.
+     * @param  payersMerkleRoot_ The payers merkle root.
+     * @param  nodeOperators_    The active node operators during the reporting period.
+     * @return digest_           The EIP-712 digest.
+     */
+    function _getPayerReportDigest(
+        uint256 originatorNodeId_,
+        uint256 attesterNodeId_,
+        uint256 startSequenceId_,
+        uint256 endSequenceId_,
+        bytes32 payersMerkleRoot_,
+        address[] calldata nodeOperators_
+    ) internal view returns (bytes32 digest_) {
+        return
+            _getDigest(
+                keccak256(
+                    abi.encode(
+                        PAYER_REPORT_TYPEHASH,
+                        originatorNodeId_,
+                        attesterNodeId_,
+                        startSequenceId_,
+                        endSequenceId_,
+                        payersMerkleRoot_,
+                        keccak256(abi.encodePacked(nodeOperators_))
+                    )
+                )
+            );
+    }
+
+    function _getRegistryParameter(bytes memory key_) internal view returns (bytes32 value_) {
+        return IParameterRegistryLike(parameterRegistry).get(key_);
+    }
+
+    function _isNotZero(address input_) internal pure returns (bool isNotZero_) {
+        return input_ != address(0);
+    }
+
+    function _toAddress(bytes32 value_) internal pure returns (address address_) {
+        // slither-disable-next-line assembly
+        assembly {
+            address_ := value_
+        }
+    }
+
+    /**
+     * @dev Verifies that enough of the signatures are valid and provided by canonical node operators and returns the
+     *      array of valid signing node IDs.
+     */
+    function _verifySignatures(
+        uint32 originatorNodeId_,
+        uint32 attesterNodeId_,
```

Comment:
I am not sure who the attester is here. All the signatures are from people attesting to the report. I'm not sure if we care who created it or who submitted it, so long as it is valid.

Reply:
I think we should ditch the field if we can't justify its existence.

Reply:
Ok, will ditch it.